File: src/main/java/org/nyxlang/interpreter/visitor/BoolVisitor.java
package org.nyxlang.interpreter.visitor;

import org.nyxlang.interpreter.IInterpreter;
import org.nyxlang.interpreter.exception.VisitorException;
import org.nyxlang.parser.ast.BoolNode;
import org.nyxlang.parser.ast.INode;


/**
 * Interprets the bool node.
 */
public class BoolVisitor implements IVisitor {

    @Override
    public boolean matches(INode node) {
        return BoolNode.class == node.getClass();
    }

    @Override
    public Boolean visit(IInterpreter interpreter, INode node) throws VisitorException {
        return ((BoolNode) node).getValue();
    }
}

File: src/main/kotlin/org/nyxlang/interpreter/visitor/BoolVisitor.kt
package org.nyxlang.interpreter.visitor

import org.nyxlang.interpreter.IInterpreter
import org.nyxlang.interpreter.result.dataResult
import org.nyxlang.interpreter.value.BoolValue
import org.nyxlang.parser.ast.BoolNode
import org.nyxlang.parser.ast.INode

/**
 * Interprets the bool node.
 */
class BoolVisitor : IVisitor {
    override fun visit(interpreter: IInterpreter, node: INode) = dataResult(BoolValue((node as BoolNode).value))
}